includes/managers/widgets.php

```diff
@@ -667,4 +663,23 @@ public function register_ajax_actions( Ajax $ajax_manager ) {
 			return $this->ajax_get_widgets_default_value_translations( $data );
 		} );
 	}
+
+	/**
+	 * @param $experiment_name
+	 * @param $classes
+	 * @return void
+	 */
+	public function register_promoted_active_widgets( $experiment_name, $classes ): void {
+		if ( ! Plugin::$instance->experiments->is_feature_active( $experiment_name ) ) {
+			return;
+		}
+
+		if ( is_array( $classes ) ) {
+			foreach ( $classes as $class_name ) {
+				$this->register( new $class_name() );
+			}
+		} else {
+			$this->register( new $classes() );
+		}
+	}
```

Comment:
Maybe, we can cast the `$classes` param to be an array then we can avoid the else condition at the bottom of this function
```suggestion
			public function register_promoted_active_widgets( string $experiment_name, array $classes ): void {
		if ( ! Plugin::$instance->experiments->is_feature_active( $experiment_name ) ) {
			return;
		}

		foreach ( $classes as $class_name ) {
			$this->register( new $class_name() );
		}
	}
```